README.md

```diff
@@ -1,21 +1,56 @@
 # Macfly
 
-**TODO: Add description**
+This library handles Fly.io Macaroon encoding, decoding, attentuation, etc..
+
+[Hex Docs](https://hexdocs.pm/macfly/)
 
 ## Installation
 
-If [available in Hex](https://hex.pm/docs/publish), the package can be installed
-by adding `macfly` to your list of dependencies in `mix.exs`:
+The package can be installed by adding `macfly` to your list of dependencies in `mix.exs`:
 
 ```elixir
 def deps do
   [
-    {:macfly, "~> 0.1.0"}
+    {:macfly, "~> 0.2.15"}
   ]
 end
 ```
 
-Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
-and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
-be found at <https://hexdocs.pm/macfly>.
+## Usage
+
+### Encoding and Decoding
+
+```elixir
+# for example tokens, see the tests or test/vectors.json file
+token = "FlyV1 fm2_lJPE..."
+
+{:ok, [%Macfly.Macaroon{}] = macaroons} = Macfly.decode(token)
+
+token = Macfly.encode(macaroons)
+```
+
+### Attenuating a Macaroon
+
+```elixir
+# for example tokens, see the tests or test/vectors.json file
+token = "FlyV1o fm2_lJPE..."
+{:ok, [%Macfly.Macaroon{}] = macaroons} = Macfly.decode(token)
+
+caveats = [
+  %Macfly.Caveat.Organization{
+    id: 1234,
+    permission: Macfly.Action.read()
+  }
+]
+
+options = %Macfly.Options{location: "abcd"}
+          |> Macfly.Options.with_caveats([Macfly.Caveat.Organization])
```

Comment:
You shouldn't need `Macfly.Options` in most cases. The `location:` option lets you use the library for non-Fly.io tokens. Specifying caveats allows you to use custom caveat structs that aren't included in the library itself.

Reply:
in this case not specifying an options didn't change the token. the tests of Macfly use options too?

Reply:
Doesn't look like `Macfly.encode` takes options as an argument.